### PR TITLE
H-52: `graph`: Fix OpenAPI spec inconsistencies

### PR DESCRIPTION
**Pull request by indietyp**
## 🌟 What is the purpose of this PR?

While trying to generate an OpenAPI client with a new version of openapi-generator I found that the OpenAPI spec is malformed (in 6.2.0 of the generator, some critical dependencies got upgraded). Basically we exploited some bugs in the previous versions to generate the typescript client 😅 

This included:

* `components.schemas.FilterExpression` referencing `#/components/schemas/Any` (which did not exist)
* `components.schemas.CreatePropertyTypeRequest`, referencing `#/components/schemas/property_type` (which did not exist)
* `components.schemas.CreateDataTypeRequest`, referencing `#/components/schemas/data_type` (which did not exist)
* `components.schemas.CreateEntityTypeRequest`, referencing `#/components/schemas/entity_type` (which did not exist)

Thanks to @TimDiekmann for guidance as to how to solve these issues!

As a drive-by, I upgraded the client generator to `6.6.0` (the latest version).

## 🔍 What does this change?

* `#/components/schemas/Any` has been removed, in favor of just allowing the value
* `*_type` now reference to their respective json schemas in the `./models` folder

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs [hunk 1]
```diff
@@ -19,9 +19,15 @@ pub enum ListOrValue<T> {
 // Utoipa doesn't seem to be able to generate sensible interfaces for this, it gets confused by
 // the generic
 impl<T> ListOrValue<T> {
-    pub(crate) fn generate_schema(schema_name: &'static str) -> RefOr<Schema> {
+    pub(crate) fn generate_schema(schema_name: &'static str, replace: bool) -> RefOr<Schema> {
```
Comment (TimDiekmann) [inline apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs:22]: Would you mind replacing the boolean with an enumeration?

File: apps/hash-graph/lib/graph/src/api/rest.rs [hunk 2]
```diff
@@ -474,13 +474,17 @@ struct FilterSchemaAddon;
...
     #[expect(clippy::too_many_lines)]
     fn modify(&self, openapi: &mut openapi::OpenApi) {
-        // This magically generates `any`, which is the closest representation we found working
-        // with the OpenAPI generator.
-        #[derive(Serialize, ToSchema)]
-        #[serde(untagged)]
-        #[expect(dead_code)]
-        enum Any {
-            Object(HashMap<String, Self>),
+        // This is a bit of hack, but basically, it adds a schema that is equivalent to "any value"
```
Comment (TimDiekmann) [inline apps/hash-graph/lib/graph/src/api/rest.rs:477]: This sounds like a comment from Copilot 😂

**Review by TimDiekmann — COMMENTED** [review]
Thanks a lot, I'm very happy to see this fixed!

I have one comment but otherwise, it looks good. Happy to approve if CI passes and the comment is addressed 🚀 ✔️

File: apps/hash-graph/lib/graph/src/api/rest.rs (hunk 2 above)
Comment (indietyp) [reply apps/hash-graph/lib/graph/src/api/rest.rs:477]: aw man. I failed the turing test ;-;

Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/api/rest.rs:477]: To be fair, Copilot would probably phrase it like
> This is a bit of a hack, but we cannot do much about it.

File: apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs (hunk 1 above)
Comment (indietyp) [reply apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs]: fixed in [4e2f303](https://github.com/hashintel/hash/pull/2718/commits/4e2f3033506bbf26a069f1964e613f0b3b43bf0c)

Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs]: Cool, thanks. Now make it `Copy`, pass it by value, and I'll approve 🪨

Comment (indietyp) [reply apps/hash-graph/lib/graph/src/api/rest/utoipa_typedef.rs]: yes sir 🫡 [d007743](https://github.com/hashintel/hash/pull/2718/commits/d007743d7f466d2fb3d8744730dca548b9e1d623)

**Review by TimDiekmann — APPROVED** [review]
Thank you!

This will result in the following diff in the `hash-graph-client` package:

<details>
<summary>Diff</summary>

```diff
diff --git a/libs/@local/hash-graph-client/api.ts b/libs/@local/hash-graph-client/api.ts
index c714ccf0d..f31efcf9d 100644
--- a/libs/@local/hash-graph-client/api.ts
+++ b/libs/@local/hash-graph-client/api.ts
@@ -13,16 +13,18 @@
  */
 
 
-import { Configuration } from './configuration';
-import globalAxios, { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+import type { Configuration } from './configuration';
+import type { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+import globalAxios from 'axios';
 // URLSearchParams not necessarily used
 // @ts-ignore
 import { URL, URLSearchParams } from 'url';
 // Some imports not used depending on template conditions
 // @ts-ignore
 import { DUMMY_BASE_URL, assertParamExists, setApiKeyToObject, setBasicAuthToObject, setBearerAuthToObject, setOAuthToObject, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from './common';
+import type { RequestArgs } from './base';
 // @ts-ignore
-import { BASE_PATH, COLLECTION_FORMATS, RequestArgs, BaseAPI, RequiredError } from './base';
+import { BASE_PATH, COLLECTION_FORMATS, BaseAPI, RequiredError } from './base';
 
 /**
  * 
@@ -1151,6 +1153,8 @@ export interface KnowledgeGraphToKnowledgeGraphOutwardEdge {
      */
     'rightEndpoint': EntityIdWithInterval;
 }
+
+
 /**
  * 
  * @export
@@ -1176,6 +1180,8 @@ export interface KnowledgeGraphToOntologyOutwardEdge {
      */
     'rightEndpoint': OntologyTypeVertexId;
 }
+
+
 /**
  * @type KnowledgeGraphVertex
  * @export
@@ -1409,6 +1415,8 @@ export interface OntologyToKnowledgeGraphOutwardEdge {
      */
     'rightEndpoint': EntityIdWithInterval;
 }
+
+
 /**
  * 
  * @export
@@ -1434,6 +1442,8 @@ export interface OntologyToOntologyOutwardEdge {
      */
     'rightEndpoint': OntologyTypeVertexId;
 }
+
+
 /**
  * 
  * @export
@@ -1539,11 +1549,17 @@ export interface ParameterExpression {
 export interface PathExpression {
     /**
      * 
-     * @type {Array<DataTypeQueryToken | PropertyTypeQueryToken | EntityTypeQueryToken | EntityQueryToken | Selector | string | number>}
+     * @type {Array<PathExpressionPathInner>}
      * @memberof PathExpression
      */
-    'path': Array<DataTypeQueryToken | PropertyTypeQueryToken | EntityTypeQueryToken | EntityQueryToken | Selector | string | number>;
+    'path': Array<PathExpressionPathInner>;
 }
+/**
+ * @type PathExpressionPathInner
+ * @export
+ */
+export type PathExpressionPathInner = DataTypeQueryToken | EntityQueryToken | EntityTypeQueryToken | PropertyTypeQueryToken | Selector | number | string;
+
 /**
  * 
  * @export
@@ -1927,6 +1943,8 @@ export interface QueryTemporalAxesDecisionTimePinned {
      */
     'timestamp': string;
 }
+
+
 /**
  * 
  * @export
@@ -1946,6 +1964,8 @@ export interface QueryTemporalAxesDecisionTimeVariable {
      */
     'interval': RightBoundedTemporalInterval;
 }
+
+
 /**
  * 
  * @export
@@ -1984,6 +2004,8 @@ export interface QueryTemporalAxesTransactionTimePinned {
      */
     'timestamp': string;
 }
+
+
 /**
  * 
  * @export
@@ -2003,6 +2025,8 @@ export interface QueryTemporalAxesTransactionTimeVariable {
      */
     'interval': RightBoundedTemporalInterval;
 }
+
+
 /**
  * @type QueryTemporalAxesUnresolved
  * Defines the two possible combinations of pinned/variable temporal axes that are used in queries that return [`Subgraph`]s.  The [`VariableTemporalAxisUnresolved`] is optionally bounded, in the absence of provided bounds an inclusive bound at the timestamp at point of resolving is assumed.  [`Subgraph`]: crate::subgraph::Subgraph
@@ -2048,6 +2072,8 @@ export interface QueryTemporalAxesUnresolvedDecisionTimePinned {
      */
     'timestamp': string | null;
 }
+
+
 /**
  * 
  * @export
@@ -2067,6 +2093,8 @@ export interface QueryTemporalAxesUnresolvedDecisionTimeVariable {
      */
     'interval': UnresolvedRightBoundedTemporalInterval;
 }
+
+
 /**
  * 
  * @export
@@ -2105,6 +2133,8 @@ export interface QueryTemporalAxesUnresolvedTransactionTimePinned {
      */
     'timestamp': string | null;
 }
+
+
 /**
  * 
  * @export
@@ -2124,6 +2154,8 @@ export interface QueryTemporalAxesUnresolvedTransactionTimeVariable {
      */
     'interval': UnresolvedRightBoundedTemporalInterval;
 }
+
+
 /**
  * 
  * @export
@@ -2257,6 +2289,8 @@ export interface Status {
      */
     'contents': Array<StatusContentsInner>;
 }
+
+
 /**
  * The canonical status codes for software within the HASH ecosystem. Sometimes multiple status codes may apply. Services should return the most specific status code that applies. For example, prefer `OutOfRange` over `FailedPrecondition` if both codes apply. Similarly prefer `NotFound` or `AlreadyExists` over `FailedPrecondition`.
  * @export
diff --git a/libs/@local/hash-graph-client/base.ts b/libs/@local/hash-graph-client/base.ts
index fab2a0ea4..1fa671137 100644
--- a/libs/@local/hash-graph-client/base.ts
+++ b/libs/@local/hash-graph-client/base.ts
@@ -13,10 +13,11 @@
  */
 
 
-import { Configuration } from "./configuration";
+import type { Configuration } from './configuration';
 // Some imports not used depending on template conditions
 // @ts-ignore
-import globalAxios, { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+import type { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+import globalAxios from 'axios';
 
 export const BASE_PATH = "http://localhost".replace(/\/+$/, "");
 
@@ -64,8 +65,8 @@ export class BaseAPI {
  * @extends {Error}
  */
 export class RequiredError extends Error {
-    name: "RequiredError" = "RequiredError";
     constructor(public field: string, msg?: string) {
         super(msg);
+        this.name = "RequiredError"
     }
 }
diff --git a/libs/@local/hash-graph-client/common.ts b/libs/@local/hash-graph-client/common.ts
index a67d69e28..c55cfb871 100644
--- a/libs/@local/hash-graph-client/common.ts
+++ b/libs/@local/hash-graph-client/common.ts
@@ -13,9 +13,10 @@
  */
 
 
-import { Configuration } from "./configuration";
-import { RequiredError, RequestArgs } from "./base";
-import { AxiosInstance, AxiosResponse } from 'axios';
+import type { Configuration } from "./configuration";
+import type { RequestArgs } from "./base";
+import type { AxiosInstance, AxiosResponse } from 'axios';
+import { RequiredError } from "./base";
 import { URL, URLSearchParams } from 'url';
 
 /**
@@ -84,24 +85,35 @@ export const setOAuthToObject = async function (object: any, name: string, scope
     }
 }
 
+function setFlattenedQueryParams(urlSearchParams: URLSearchParams, parameter: any, key: string = ""): void {
+    if (parameter == null) return;
+    if (typeof parameter === "object") {
+        if (Array.isArray(parameter)) {
+            (parameter as any[]).forEach(item => setFlattenedQueryParams(urlSearchParams, item, key));
+        } 
+        else {
+            Object.keys(parameter).forEach(currentKey => 
+                setFlattenedQueryParams(urlSearchParams, parameter[currentKey], `${key}${key !== '' ? '.' : ''}${currentKey}`)
+            );
+        }
+    } 
+    else {
+        if (urlSearchParams.has(key)) {
+            urlSearchParams.append(key, parameter);
+        } 
+        else {
+            urlSearchParams.set(key, parameter);
+        }
+    }
+}
+
 /**
  *
  * @export
  */
 export const setSearchParams = function (url: URL, ...objects: any[]) {
     const searchParams = new URLSearchParams(url.search);
-    for (const object of objects) {
-        for (const key in object) {
-            if (Array.isArray(object[key])) {
-                searchParams.delete(key);
-                for (const item of object[key]) {
-                    searchParams.append(key, item);
-                }
-            } else {
-                searchParams.set(key, object[key]);
-            }
-        }
-    }
+    setFlattenedQueryParams(searchParams, objects);
     url.search = searchParams.toString();
 }
```

</details>

**Comment by indietyp** [comment]
Look like quite the improvement, nice that it now uses `type` imports!

**Pull request merged**